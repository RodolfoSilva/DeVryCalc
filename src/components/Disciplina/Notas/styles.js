import { colors } from '../../../styles'

export const styleByNota = (nota = null) => {
  if (nota === null) return {}
  if (nota >= 5) return { color: colors.secondary }
  return { color: colors.danger }
}
