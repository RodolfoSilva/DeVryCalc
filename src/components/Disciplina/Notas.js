import React from 'react'
import { View } from 'react-native'
import Nota from './Nota'
import { colors } from '../../styles'
import { styleByNota } from './styles'

const Notas = ({ ap1, ap2, ap3, style }) => (
  <View style={style}>
    <Nota value={ap1} valueStyle={styleByNota(ap1)} title="AP1" />
    <Nota value={ap2} valueStyle={styleByNota(ap2)} title="AP2" />
    <Nota value={ap3} valueStyle={styleByNota(ap3)} title="AP3" />
  </View>
)

export default Notas
