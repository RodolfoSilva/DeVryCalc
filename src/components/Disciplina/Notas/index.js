import React from 'react'
import { View } from 'react-native'
import Nota from '../Nota'
import PropTypes from 'prop-types'
import { styleByNota } from './styles'

const Notas = ({ ap1 = null, ap2 = null, ap3 = null, style }) => (
  <View style={style}>
    <Nota value={ap1} valueStyle={styleByNota(ap1)} title="AP1" />
    <Nota value={ap2} valueStyle={styleByNota(ap2)} title="AP2" />
    <Nota value={ap3} valueStyle={styleByNota(ap3)} title="AP3" />
  </View>
)

Notas.propTypes = {
  ap1: PropTypes.number,
  ap2: PropTypes.number,
  ap3: PropTypes.number,
  style: View.propTypes.style
}

export default Notas
