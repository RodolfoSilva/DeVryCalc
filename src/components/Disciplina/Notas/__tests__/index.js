import React from 'react'
import Notas from '../'
import { shallow } from 'enzyme'

describe('Notas', () => {
  it('Renderiza todas as notas apenas com os placelholders', () => {
    const wrapper = shallow(<Notas />)
    expect(wrapper).toMatchSnapshot()
  })
  it('Renderiza apenas a AP1', () => {
    const wrapper = shallow(<Notas ap1={5} />)
    expect(wrapper).toMatchSnapshot()
  })
  it('Renderiza apenas a AP2', () => {
    const wrapper = shallow(<Notas ap2={3} />)
    expect(wrapper).toMatchSnapshot()
  })
  it('Renderiza apenas a AP3', () => {
    const wrapper = shallow(<Notas ap3={9} />)
    expect(wrapper).toMatchSnapshot()
  })
  it('Renderiza todas as notas', () => {
    const wrapper = shallow(<Notas ap1={4} ap2={7} ap3={5} />)
    expect(wrapper).toMatchSnapshot()
  })
})
