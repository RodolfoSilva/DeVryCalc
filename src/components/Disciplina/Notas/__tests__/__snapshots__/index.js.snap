// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notas Renderiza apenas a AP1 1`] = `
<View>
  <Nota
    title="AP1"
    value={5}
    valueStyle={
      Object {
        "color": "#32DB64",
      }
    }
  />
  <Nota
    title="AP2"
    value={null}
    valueStyle={Object {}}
  />
  <Nota
    title="AP3"
    value={null}
    valueStyle={Object {}}
  />
</View>
`;

exports[`Notas Renderiza apenas a AP2 1`] = `
<View>
  <Nota
    title="AP1"
    value={null}
    valueStyle={Object {}}
  />
  <Nota
    title="AP2"
    value={3}
    valueStyle={
      Object {
        "color": "#F53D3D",
      }
    }
  />
  <Nota
    title="AP3"
    value={null}
    valueStyle={Object {}}
  />
</View>
`;

exports[`Notas Renderiza apenas a AP3 1`] = `
<View>
  <Nota
    title="AP1"
    value={null}
    valueStyle={Object {}}
  />
  <Nota
    title="AP2"
    value={null}
    valueStyle={Object {}}
  />
  <Nota
    title="AP3"
    value={9}
    valueStyle={
      Object {
        "color": "#32DB64",
      }
    }
  />
</View>
`;

exports[`Notas Renderiza todas as notas 1`] = `
<View>
  <Nota
    title="AP1"
    value={4}
    valueStyle={
      Object {
        "color": "#F53D3D",
      }
    }
  />
  <Nota
    title="AP2"
    value={7}
    valueStyle={
      Object {
        "color": "#32DB64",
      }
    }
  />
  <Nota
    title="AP3"
    value={5}
    valueStyle={
      Object {
        "color": "#32DB64",
      }
    }
  />
</View>
`;

exports[`Notas Renderiza todas as notas apenas com os placelholders 1`] = `
<View>
  <Nota
    title="AP1"
    value={null}
    valueStyle={Object {}}
  />
  <Nota
    title="AP2"
    value={null}
    valueStyle={Object {}}
  />
  <Nota
    title="AP3"
    value={null}
    valueStyle={Object {}}
  />
</View>
`;
